core/src/main/java/com/linecorp/armeria/client/retry/HttpStatusBasedRetryStrategy.java

```diff
@@ -36,12 +36,16 @@
 
     private final Function<HttpStatus, Optional<Backoff>> statusBasedBackoffFunction;
 
+    private Backoff backoffOnException;
+
     /**
      * Creates a new instance.
      */
-    HttpStatusBasedRetryStrategy(Function<HttpStatus, Optional<Backoff>> statusBasedBackoffFunction) {
+    HttpStatusBasedRetryStrategy(Function<HttpStatus, Optional<Backoff>> statusBasedBackoffFunction,
+                                 Backoff backoffOnException) {
```

Comment:
Shouldn't this be `Function<? super Exception, Optional<Backoff>> exceptionBasedBackoffFunction`?